Homework3 fragments

Voodstr opened the pull request.

Commit:
README.md Домашка №3

Commit:
Добавлена реализация bottom_navigation для переключения между фрагментами
Три фрагмента FilmListFragment.kt, FavoriteFilmsFragment.kt, SearchFragment.kt

Commit:
Добавлена реализация bottom_navigation для переключения между фрагментами
Три фрагмента FilmListFragment.kt, FavoriteFilmsFragment.kt, SearchFragment.kt

Commit:
Исправлена ошибка в FavoriteFilmsFragment.kt

Commit:
Добавлена поддержка AboutFragment.kt

Commit:
1. Toast при добавление(удаление) в избранное
2. Анимация при переходе в "Подробнее"

Commit:
1. Title не должен вылезать теперь
2. Обновлен список фильмов

Commit:
Обновлено
CoordinatorLayout added
Collapsing toolbar added

baidala commented on app/src/main/AndroidManifest.xml:
```diff
@@ -10,7 +10,6 @@
         android:supportsRtl="true"
         android:theme="@style/Theme.OtusLesson">
-        <activity android:name=".FavoriteFilmsActivity"></activity>
         <activity android:name=".AboutActivity" />
```
этого уже нет

baidala commented on app/src/main/java/ru/voodster/otuslesson/fragment/FilmListFragment.kt:
```diff
@@ -0,0 +1,48 @@
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        if(activity is OnFilmClickListener){
```
инициализацию лиснера (которым будет активити) лучше делать в onAttach()

baidala commented on app/src/main/java/ru/voodster/otuslesson/film/FilmAdapter.kt:
```diff
@@ -20,8 +23,6 @@ class FilmAdapter(private val filmsList : ArrayList<FilmItem>) : RecyclerView.Ad
-        // Log.d(TAG,"onCreateViewHolder $viewType")
-        val layoutInflater = LayoutInflater.from(parent.context)
         return if (viewType==0) {
                 val view = layoutInflater.inflate(R.layout.item_film, parent, false)
```
рекомендую делать  через LayoutInflater.from(parent.context).inflate( 
и  не надо через конструкор прокидывать

baidala reviewed (changes requested):
хорошая работа. исправьте небольшие замечания и можно двигатся дальше

Commit:
исправлены ошибки,
удаление из списка добавлено, но удаление одноразовое т.к. item меняет позицию а значит и позицию он должен получать новую

Commit:
последний элемент из списка всегда out of bounds

Commit:
есть метод removeLast() только не пойму почему по номеру элемента это сделать было нельзя

Commit:
Исправил баг с отметкой избранного.

Commit:
Добавлен диалог подтверждения удаления

Commit:
Добавлен диалог подтверждения удаления

Commit:
animation fix
